processor/deltatorateprocessor/processor.go

```diff
@@ -74,8 +74,14 @@ func (dtrp *deltaToRateProcessor) processMetrics(_ context.Context, md pdata.Met
 					fromDataPoint.CopyTo(newDp)
 
 					durationNanos := time.Duration(fromDataPoint.Timestamp() - fromDataPoint.StartTimestamp())
-					rate := calculateRate(fromDataPoint.DoubleVal(), durationNanos)
-					newDp.SetDoubleVal(rate)
+					switch fromDataPoint.Type() {
+					case pdata.MetricValueTypeDouble:
+						rate := calculateRate(fromDataPoint.DoubleVal(), durationNanos)
+						newDp.SetDoubleVal(rate)
+					case pdata.MetricValueTypeInt:
+						rate := calculateRate(float64(fromDataPoint.IntVal()), durationNanos)
+						newDp.SetDoubleVal(rate)
```

Comment:
Are those the only possible data point types? Would it make sense to have a default case, in case this constraint changes in the future? If you declare the rate outside of the switch, you can simplify this by having the `newDp.SetDoubleVal` being specified only once, after the switch.

Reply:
Simplify code done by latest commit. Thank you.